### PR TITLE
Add GRPCRoute to Terminate-supported types

### PR DESCRIPTION
**Pull request by rainest**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Add GRPCRoute to the list of route types supported by TLS Terminate mode for GRPCS.

It looks like this should be supported but wasn't updated when GRPCRoute was added.

**Does this PR introduce a user-facing change?**:

```release-note
Added GRPCRoute to list of route types supported by TLS Terminate mode.
```

### CODE REVIEW

File: site-src/guides/tls.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ Listener Protocol | TLS Mode | Route Type Supported
 TLS | Passthrough | TLSRoute
 TLS | Terminate | TCPRoute
 HTTPS | Terminate | HTTPRoute
+GRPCS | Terminate | GRPCRoute
```
Comment (robscott) [inline site-src/guides/tls.md:33]:
```suggestion
GRPC | Terminate | GRPCRoute
```

**Review by robscott — APPROVED** [review]
Thanks @rainest! Just one small nit.

**Review by shaneutt — APPROVED** [review]
/approve
/lgtm

**Comment by youngnick** [comment]
Updates #1016

**Pull request merged**